keps/sig-cluster-lifecycle/generic/1755-communicating-a-local-registry/README.md

```diff
@@ -0,0 +1,655 @@
+  interact with it. They think deleting the ConfigMap would delete the local
+  registry (which it does not).
+
+- This KEP does not specify any automatic mechanism for keeping the
+  LocalRegistryHosting up-to-date with how the cluster is configured.
+  For example, the user might delete the registry. The cluster might
+  not have a way of knowing that the registry has died.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the SIG or subproject.
+-->
+
+None
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable.  This may include API specs (though not always
+required) or even code snippets.  If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### The `local-registry-hosting` ConfigMap
+
+Tools that configure a local registry should apply a ConfigMap to the cluster.
+
+Documentation that educates people on how to set up a registry manually should
+include instructions on what the ConfigMap should be.
+
+The name of the ConfigMap must be `local-registry-hosting`.
+
+The namespace must be `kube-public`.
+
+Under `data`, the ConfigMap should contain structures in the format
+`localRegistryHosting.vX`.  `vX` is the major version of the structure. The
+contents of each field are YAML. This proposal only specifies `v1`.
+
+Example:
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: local-registry-hosting
+  namespace: kube-public
+```
+
+### LocalRegistryHosting
+
+The `localRegistryHosting.vX` field of the ConfigMap describes how tools should
+communicate with a local registry. Tools should use this registry to load images
+into the cluster.
+
+The contents of a `localRegistryHosting.vX` specification are frozen and will not
+change.
+
+When adding, removing or renaming fields, future proposals should increment the
+MAJOR version of an API hosted in the ConfigMap - e.g. `localRegistryHosting.vX`
+where X is incremented.
+
+Writers of this ConfigMap (i.e., cluster configuration tooling) should write as
+many top-level fields as the cluster supports.  The most recent version is the
+source of truth.
+
+Readers of this ConfigMap should start with the most recent version they support
+and work backwards. Readers are responsible for doing any defaulting of fields
+themselves without the assistance of any common API machinery.
+
+#### Specification for LocalRegistryHosting v1
+
+Example ConfigMap:
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: local-registry-hosting
+  namespace: kube-public
+data:
+  localRegistryHosting.v1: |
+    host: "localhost:5000"
```

Comment:
i'd assume we want to add hostFromClusterNetwork here too?

Reply:
done